transport/rif_comms/node.py

```diff
@@ -1,135 +1,228 @@
 
 
-class RifCommsNode(TransportNode):
+class Node(TransportNode):
     _log = log
 
     def __init__(self, address: Address, config: dict):
         TransportNode.__init__(self, address)
+
+        # set instance variables
         self._config = config
         self._raiden_service: RaidenService = None
 
+        # set up comms node
         self._rif_comms_connect_stream: Notification = None
-        self._our_topic: Notification = None
+        self._our_topic_stream: Notification = None
         self._our_topic_thread: Greenlet = None
+        self._comms_client = RIFCommsClient(to_checksum_address(address), self._config["grpc_endpoint"])
+        print("RIFCommsNode init on GRPC endpoint: {}".format(self._config["grpc_endpoint"]))
 
-        self._client = RifCommsClient(to_checksum_address(address), self._config["grpc_endpoint"])
-        print("RifCommsNode init on grpc endpoint: {}".format(self._config["grpc_endpoint"]))
-
-        self._greenlets: List[Greenlet] = list()
+        # initialize message queues
         self._address_to_message_queue: Dict[Address, MessageQueue] = dict()
 
-        self._stop_event = Event()
-        self._stop_event.set()
-
         self._log = log.bind(node_address=pex(self.address))
 
-    @property
-    def _queueids_to_queues(self) -> QueueIdsToQueues:
-        chain_state = views.state_from_raiden(self._raiden_service)
-        return views.get_all_messagequeues(chain_state)
-
     def start(self, raiden_service: RaidenService, message_handler: MessageHandler, prev_auth_data: str):
-        if not self._stop_event.ready():
+        """
+        Initialize transport fields, subscribe to own topic in order to listen for messages, and start as Runnable.
+        """
+        self._raiden_service = raiden_service  # TODO: this should be set in __init__
+
+        # check if node is already running
+        if not self.stop_event.ready():
             raise RuntimeError(f"{self!r} already started")
-        self._stop_event.clear()
-        self._raiden_service = raiden_service
+        self.stop_event.clear()
+
+        # connect to rif comms node
+        # TODO: this shouldn't need to be assigned, it is only done because otherwise the code hangs
+        self._rif_comms_connect_stream = self._comms_client.connect()
 
-        self._rif_comms_connect_stream = self._client.connect()
-        self._our_topic = self._client.subscribe(to_checksum_address(raiden_service.address))
-        self._client.get_peer_id(
-            to_checksum_address(raiden_service.address))  # TODO remove this after grpc api request blocking is fixed
+        # subscribe to our own topic to receive messages
+        self._start_message_listener()
+
+        # start pre-loaded message queues
         for message_queue in self._address_to_message_queue.values():
-            if not message_queue:
-                self.log.debug("Starting message_queue", message_queue=message_queue)
+            if not message_queue.greenlet:
```

Comment:
i dont understand why checking the greenlet attribute

Reply:
see [here](https://github.com/rsksmart/lumino/pull/161#discussion_r527697926)

Reply:
otherwise;

```python
for message_queue in values:
    if not message_queue:
```

makes no sense